src/OmniSharp.DotNetTest/TestFrameworks/TestFramework.cs

```diff
@@ -0,0 +1,71 @@
+﻿using System.Collections.Generic;
+using System.Collections.Immutable;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+
+namespace OmniSharp.DotNetTest.TestFrameworks
+{
+    public abstract class TestFramework
+    {
+        private static readonly ImmutableDictionary<string, TestFramework> s_frameworks;
+
+        static TestFramework()
+        {
+            var builder = ImmutableDictionary.CreateBuilder<string, TestFramework>();
+
+            var nunit = new NUnitTestFramework();
+            var xunit = new XunitTestFramework();
+
+            builder.Add(nunit.Name, nunit);
+            builder.Add(xunit.Name, xunit);
+
+            s_frameworks = builder.ToImmutable();
+        }
+
+        public static TestFramework GetFramework(string name)
+        {
+            return s_frameworks[name];
```

Comment:
I think this will throw a `KeyNotFoundException` before you get a chance to check it for null here https://github.com/OmniSharp/omnisharp-roslyn/pull/808/files#diff-54ef29f56e5eaa19dc93f2f6b451ee05R47